### PR TITLE
limit information in CLI error printing

### PR DESCRIPTION
**Pull request by elharo**
@suztomo The goal here is not to dump a stack trace or other random content, just print what actually went wrong. Also this uses stderr instead of stdout.

### CODE REVIEW

File: dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerArguments.java [hunk 1]
```diff
@@ -84,9 +82,6 @@ static LinkageCheckerArguments readCommandLine(String... arguments) throws Parse
       return new LinkageCheckerArguments(parser.parse(options, arguments));
     } catch (IllegalArgumentException ex) {
       throw new ParseException("Invalid URL syntax in Maven repository URL");
-    } catch (ParseException ex) {
-      helpFormatter.printHelp("LinkageChecker", options);
```
Comment (suztomo) [inline dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerArguments.java]: Does the command still shows available options when invoked without any argument?

Comment (elharo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerArguments.java]: It does not. We should distinguish help from an error condition. The most common convention seems to be that -h or --help prints help.

Comment (suztomo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerArguments.java]: Common behaviors print the help when a required argument is missing. Would you update this PR to do so?

Comment (elharo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerArguments.java]: I added a --help/-h option. With that there are no required arguments any more.

Also, I think there's an existing bug where jar files can't be selected.

Comment (suztomo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerArguments.java]: I just ran LinkageCheckerMain without any argument and it existed 0 without showing anything. The empty output does not tell what option I have to add LinkgeCheckerMain.  Would you fix that?

Comment (suztomo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerArguments.java]: > Common behaviors print the help when a required argument is missing.

I'm referring `java` or `javac` command without any argument, for example.

Comment (elharo) [reply dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageCheckerArguments.java]: I'll see your javac and raise you a gcc:

```
$ gcc
gcc: fatal error: no input files
compilation terminated.
```

but sure, whatever.

**Pull request merged**